File: Lampada.java
public class Lampada {
    private String modelo;
    private String tensao;
    private int potencia;
    private String cor;
    private String tipoLuz;
    private int garantiaMeses;
    private boolean ligada;

    // Construtor
    public Lampada(String modelo, String tensao, int potencia, String cor, String tipoLuz, int garantiaMeses) {
        this.modelo = modelo;
        this.tensao = tensao;
        this.potencia = potencia;
        this.cor = cor;
        this.tipoLuz = tipoLuz;
        this.garantiaMeses = garantiaMeses;
        this.ligada = false; // inicia desligada
    }

    // Métodos
    public void ligar() {
        ligada = true;
        System.out.println("A lâmpada foi ligada.");
    }

    public void desligar() {
        ligada = false;
        System.out.println("A lâmpada foi desligada.");
    }

    public void mostrarEstado() {
        if (ligada) {
            System.out.println("A lâmpada " + modelo + " está LIGADA.");
        } else {
            System.out.println("A lâmpada " + modelo + " está DESLIGADA.");
        }
    }
}

File: lampada.js
class Lampada {
    constructor(modelo, tensao, potencia, cor, tipoLuz, garantiaMeses) {
        this.modelo = modelo;          // String
        this.tensao = tensao;          // String
        this.potencia = potencia;      // int
        this.cor = cor;                // String
        this.tipoLuz = tipoLuz;        // String
        this.garantiaMeses = garantiaMeses; // int
        this.ligada = false;           // boolean
    }

    ligar() {
        this.ligada = true;
        console.log("A lâmpada foi ligada.");
    }

    desligar() {
        this.ligada = false;
        console.log("A lâmpada foi desligada.");
    }

    mostrarEstado() {
        if (this.ligada) {
            console.log(`A lâmpada ${this.modelo} está LIGADA.`);
        } else {
            console.log(`A lâmpada ${this.modelo} está DESLIGADA.`);
        }
    }
}


const lampadaLed = new Lampada("Philips", "110V", 20, "Branca", "LED", 12);

lampadaLed.mostrarEstado(); // Desligada
lampadaLed.ligar();
lampadaLed.mostrarEstado(); // Ligada
lampadaLed.desligar();
lampadaLed.mostrarEstado(); // Desligada


